os/fs/vfs/fs_stat.c

```diff
@@ -241,3 +245,81 @@ int stat(FAR const char *path, FAR struct stat *buf)
 	set_errno(ret);
 	return ERROR;
 }
+
+/****************************************************************************
+ * Name: inode_stat
+ *
+ * Description:
+ *   The inode_stat() function will obtain information about an 'inode' in
+ *   the pseudo file system and will write it to the area pointed to by 'buf'.
+ *
+ *   The 'buf' argument is a pointer to a stat structure, as defined in
+ *   <sys/stat.h>, into which information is placed concerning the file.
+ *
+ * Input Parameters:
+ *   inode - The inode of interest
+ *   buf   - The caller provide location in which to return information about
+ *           the inode.
+ *
+ * Returned Value:
+ *   Zero (OK) returned on success.  Otherwise, a negated errno value is
+ *   returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+
+int inode_stat(FAR struct inode *inode, FAR struct stat *buf)
```

Comment:
This is same as statpseudo, let me know why inode_stat added?

Reply:
Hmm, they look similar, will try to dedup them.

Reply:
I mean inode_stat's code is same as statpesudo.

Reply:
The only significant difference is in line 98:
> $git blame | head -100 | tail -4
>4fc385da (junmin.kim     2016-12-23 02:13:53 -0800  96) #if defined(CONFIG_FS_SHM)
>4fc385da (junmin.kim     2016-12-23 02:13:53 -0800  97)                 if (INODE_IS_SHM(inode)) {
>4fc385da (junmin.kim     2016-12-23 02:13:53 -0800  98)                         buf->st_mode | S_IFSHM;
>4fc385da (junmin.kim     2016-12-23 02:13:53 -0800  99)                 }

In new code it is:
>08247b90 (Tomasz Wozniak 2017-11-02 10:08:36 +0100 291) 			buf->st_mode = S_IFSHM;

Not sure what the author of the 4fc385da had in mind, but I guess it is just a bug, and I'll act accordingly. Thanks for spotting the similarity.

Reply:
Almost, but not the same. Removed statpseudo.